pkg/instrumentation/sdk.go

```diff
@@ -143,6 +143,16 @@ func (i *sdkInjector) inject(ctx context.Context, insts languageInstrumentations
 		pod = i.injectCommonEnvVar(otelinst, pod, index)
 		pod = i.injectCommonSDKConfig(ctx, otelinst, ns, pod, index, index)
 	}
+	if insts.Nginx != nil {
+		otelinst := *insts.Nginx
+		i.logger.V(1).Info("injecting Nginx instrumentation into pod", "otelinst-namespace", otelinst.Namespace, "otelinst-name", otelinst.Name)
+		// Nginx agent is configured via config files rather than env vars.
+		// Therefore, service name, otlp endpoint and other attributes are passed to the agent injection method
+		pod = injectNginxSDK(i.logger, otelinst.Spec.Nginx, pod, index, otelinst.Spec.Endpoint, i.createResourceMap(ctx, otelinst, ns, pod, index))
+		pod = i.injectCommonEnvVar(otelinst, pod, index)
+		pod = i.injectCommonSDKConfig(ctx, otelinst, ns, pod, index, index)
+		i.logger.V(1).Info("injecting Nginx instrumentation into pod", "pod", pod, "volumes", pod.Spec.Volumes)
+	}
```

Comment:
Adding this comment here since GitHub doesn't allow me to put it in another place.

Set the `securityContext` using the `setInitContainerSecurityContext` for the new `initContainer`. (See [#2051](https://github.com/open-telemetry/opentelemetry-operator/pull/2051))

Reply:
It looks this is done for all initContainers for all languages - is there a specific change needed?

Reply:
It is not. If you check [this change I'm proposing as part of](https://github.com/open-telemetry/opentelemetry-operator/blob/6fcd5323b16bfcecd23cbd9b81a4f336c3fa83ff/pkg/instrumentation/sdk.go#L161C40-L161C40) #2051, you will see that the current code just checks for the `opentelemetry-auto-instrumentation` container. In your case, you need to check for the `nginxAgentInitContainerName` if I'm not wrong.

Reply:
Oh, got it. Sorry for that. Fixed. Also added specific e2e test for container-level securityContext.